tests/meshcop/commissioner.hpp

```diff
@@ -58,241 +52,175 @@
-
-            /** ascii & binary of PSKc, either from calculation or cmdline */
-            char    ascii[(OT_PSKC_LENGTH * 2) + 1];
-            uint8_t bin[OT_PSKC_LENGTH];
-        } mPSKc;
-
-    } mAgent;
-
-    /** Kek with the joiner operation */
-    uint8_t mKek[32];
-
-    struct comm_ka
+public:
+    /**
+     * The constructor to initialize Commissioner
+     *
+     * @param[in]    aPskcBin           binary form of pskc
+     * @param[in]    aPskdAscii         ascii form of pskd
+     * @param[in]    aSteeringData      default steering data to filter joiner
+     * @param[in]    aKeepAliveRate     send keep alive packet every aKeepAliveRate seconds
+     *
+     */
+    Commissioner(const uint8_t *     aPskcBin,
+                 const char *        aPskdAscii,
+                 const SteeringData &aSteeringData,
+                 int                 aKeepAliveRate);
+
+    /**
+     * This method updates the fd_set and timeout for mainloop.
+     * @p aTimeout should only be updated if session has pending process in less than its current value.
+     *
+     * @param[inout]    aReadFdSet      A reference to fd_set for polling read.
+     * @param[inout]    aWriteFdSet     A reference to fd_set for polling write.
+     * @param[inout]    aErrorFdSet     A reference to fd_set for polling error.
+     * @param[inout]    aMaxFd          A reference to the current max fd in @p aReadFdSet and @p aWriteFdSet.
+     * @param[inout]    aTimeout        A reference to the timeout.
+     *
+     */
+    void UpdateFdSet(fd_set &aReadFdSet, fd_set &aWriteFdSet, fd_set &aErrorFdSet, int &aMaxFd, timeval &aTimeout);
+
+    /**
+     * This method performs the session processing.
+     *
+     * @param[in]   aReadFdSet          A reference to fd_set ready for reading.
+     * @param[in]   aWriteFdSet         A reference to fd_set ready for writing.
+     * @param[in]   aErrorFdSet         A reference to fd_set with error occurred.
+     *
+     */
+    void Process(const fd_set &aReadFdSet, const fd_set &aWriteFdSet, const fd_set &aErrorFdSet);
+
+    /**
+     * This method returns whether the commissioner is valid
+     *
+     * @returns inner commisioner state is not kStateInvalid
+     *
+     */
+    bool IsValid();
+
+    /**
+     * This method initialize the dtls session
+     *
+     * @param[in]   aAgentAddr      Address of border agent
+     *
+     * @returns 0 on success, MBEDTLS_ERR_XXX on failure
+     *
+     */
+    int InitDtls(const sockaddr_in &aAgentAddr);
+
+    /**
+     * This method initialize the dtls session
+     *
+     * @returns 0 on success,
+     *          MBEDTLS_ERR_SSL_WANT_READ or MBEDTLS_ERR_SSL_WANT_WRITE for pending, in which case
+     *          you need to call this method again,
+     *          or other SSL error code on failure
+     */
+    int TryDtlsHandshake();
+
+    /**
+     * This method sends commissioner petition coap request
+     *
+     */
+    void CommissionerPetition();
```

Comment:
All void functions or methods shall explicitly declare and specify the void type keyword.